src/core/infer_trace.h

```diff
@@ -52,7 +53,31 @@ class InferenceTrace {
   {
   }
 
+  InferenceTrace(
```

Comment:
Make this the only constructor and change callers to explicitly pass tensor_activity_fn = nullptr when necessary